cpp/mg_utility/mg_graph.hpp

```diff
@@ -134,26 +132,39 @@ class Graph : public GraphView<TSize> {
   /// Creates an edge.
   ///
   /// Creates an undirected edge in the graph, but edge will contain information
-  /// about the original directed property.
+  /// about the original directed property. Throws exception if node id's are not contained in  graph.
   ///
   /// @param[in]  from  The from node identifier
   /// @param[in]  to    The to node identifier
   ///
   /// @return     Created edge id
-  TSize CreateEdge(std::uint64_t memgraph_id_from,
-                   std::uint64_t memgraph_id_to) {
-    auto from = memgraph_to_inner_id_[memgraph_id_from];
-    auto to = memgraph_to_inner_id_[memgraph_id_to];
+  TSize CreateUndirectedEdge(std::uint64_t memgraph_id_from, std::uint64_t memgraph_id_to) {
```

Comment:
hmm, have a single function:
```
TSize CreateEdge(std::uint64_t from, std::uint64_t to, bool is_undirected)
```
Then you have more or less the logic from the CreateDirectedEdge function with the check at the end
```
if (is_undirected) {
  adj_list_[to]...
   ...
}
```
or if you want to be extra performance freaks (to eliminate the if), you can have compile time parameter
`template <bool is_undirected>`
and have 
```
if constexpr (is_undirected) {
  adj_list_[to]...
}
```
but then you can't eliminate the switch statements above.

Reply:
Separate functions were made on purpose. I wanted to emphasize that there exist 2 different functions for edge creation and raise awareness for the final user. 

However, since there might be more use-cases where a switch would be needed, I'll refactor this.

Reply:
@antonio2368 
For the purpose of readability and easier extension, do you think it is better to model graph type with `bool is_undirected`, or to use custom enumeration `mg_graph::GraphType graph_type`

Reply:
yeah, enum is much better as it's more explicit + you should be able to use the enum as template argument if you want.
I just don't like repeating (e.g. calling `GetInnerNodeId` twice), you can avoid that with `if`/`if constexpr`.

Reply:
I updated following your comment. I chose a solution without templatizing the CreateEdge. I wanted to eliminate all unnecessary confusion for the end-user.